lib/evaluate/intrinsics.cc

```diff
@@ -613,10 +622,13 @@ static const IntrinsicInterface genericIntrinsicFunction[]{
 //  EOF, FP_CLASS, INT_PTR_KIND, ISNAN, MALLOC
 //  probably more (these are PGI + Intel, possibly incomplete)
 
+// The following table contains the intrinsic functions listed in
+// Tables 16.2 and 16.3 in Fortran 2018.  The "unrestricted" functions
+// in Table 16.2 can be used as actual arguments, PROCEDURE() interfaces,
+// and procedure pointer targets.
 struct SpecificIntrinsicInterface : public IntrinsicInterface {
   const char *generic{nullptr};
-  bool isRestrictedSpecific{
-      false};  // when true, can only be called, not passed
+  bool isRestrictedSpecific{false};
```

Comment:
There was no functional change here, but the rewriting made me realize that both `SpecificIntrinsicInterface` and `SpecificIntrinsic` have `false` has a default value for `isRestrictedSpecific`.
This means that when `IntrinsicProcTable::Probe` finds an intrinsic in the generic table, the `specificIntrinsic` member of the returned `SpecificCall` is currently default initialized to isRestrictedSpecific{false}.
This is probably wrong if generic intrinsics that are not specific intrinsics shall not be used in a procedure pointer assignment.
I think SpecificIntrinsic.isRestrictedSpecific is not used anywhere so far, but maybe Probe should set isRestrictedSpecific to false for intrinsics that were found in the generic table just to be safe.